File: lab2_task3/main.rb
def count_sequences(sequence)
    return 0 if sequence.size < 2
    
    seq_len = 1
    seq_count = 0
    
    (0...sequence.size-1).each do |i|
      if sequence[i+1] >= sequence[i]
        seq_len += 1
      elsif seq_len > 1
        seq_count += 1
        seq_len = 1
      else
        seq_len = 1
      end
    end
    
    seq_count += 1 if seq_len > 1
    seq_count
  end
  
  print "Введите последовательность чисел через пробел: "
  nums = gets.chomp.split.map(&:to_i)
  puts count_sequences(nums)

File: lab2_task3/Main.java
import java.util.*;

public class Main {
    public static int countSequences(List<Integer> sequence) {
        if (sequence.size() < 2) return 0;
        
        int seqLen = 1;
        int seqCount = 0;
        
        for (int i = 0; i < sequence.size() - 1; i++) {
            if (sequence.get(i+1) >= sequence.get(i)) {
                seqLen++;
            } else if (seqLen > 1) {
                seqCount++;
                seqLen = 1;
            } else {
                seqLen = 1;
            }
        }
        
        if (seqLen > 1) seqCount++;
        
        return seqCount;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Введите последовательность чисел через пробел:");
        String[] input = sc.nextLine().split(" ");
        List<Integer> nums = new ArrayList<>();
        for (String s : input) {
            nums.add(Integer.parseInt(s));
        }
        System.out.println(countSequences(nums));
    }
}